doc/spec/manifest.md

```diff
@@ -0,0 +1,217 @@
+# Distribution Content Manifests
+
+A Content Manifest is a simple JSON file which contains general fields that
+are typical to any package management system. The goal is for these manifests
+to describe an application and its dependencies in a content-addressable and
+verifiable way.
+
+## Field Descriptions
+
+All fields are required unless marked `(optional)`.
+
+- **`schemaVersion`** *int*
+
+    Specifies the schema version of the manifest as an integer. This document 
+    describes version `2` only.
+
+- **`createdAt`** *string*
+
+    ISO-8601 formatted combined date and time at which the manifest was
+    created.
+
+- **`maintainers`** *array*
+
+    An array of objects representing persons or organizations which are
+    designated as mainainers of the described content. Fields are:
+    
+    - **`name`** *string*
+    
+        The name of a maintaining person or organization.
+    
+    - **`email`** *string* `(optional)`
+    
+        The email address of a maintainer.
+    
+    - **`url`** *string* `(optional)`
+    
+        A URL to a maintainer's website, blog, etc.
+
+- **`description`** *string*
+
+    A brief description of the content or application.
+
+- **`keywords`** *array*
+
+    An array of strings which should be considered keywords that may be used
+    by a separate search system to index content.
+
+- **`website`** *string*
+
+    A URL to a related website or project page for the application.
+
+- **`licenses`** *array*
+
+    An array of strings which identify any software licenses which apply
+    to the application. Values should be a [SPDX License Identifier](https://spdx.org/licenses/).
+
+- **`dependencies`** *array*
+
+    An array of objects which describe content which this application depends
+    on. Those may be generic blobs of data or other content manifests and
+    their dependencies as described by their media type. Fields are:
+    
+    - **`mediaType`** *string*
+    
+        The MIME type of the dependency object.
+    
+    - **`size`** *int*
+    
+        The size in bytes of the dependency object.
+    
+    - **`repository`** *string* `(optional)`
```

Comment:
I'm probably being dense here, so sorry :-),  but why is the target of a URL more or less stable than the target of a "repository" field?  In the end both hit a "something" and get back data.  The stability of the content will be guaranteed by checking what is downloaded against the digest - not due to the string itself. Even with a 'repo' string that has to be converted into something useable, you still have to check the data to make sure it matches the digest, right?

At least with a URL there's no magic w.r.t. how to convert it into something usable.  So, instead of having to (via written text) tell someone to convert `isv.example.com/base/system` to `https://isv.example.com/base/system`, why not just use the `https:/...` URL to begin with?  You can even still include the sha in the URL if you want.